libtsuba/src/RDGCore.cpp

```diff
@@ -164,15 +170,27 @@ RDGCore::AddEdgeProperties(const std::shared_ptr<arrow::Table>& props) {
 }
 
 katana::Result<void>
-RDGCore::UpsertNodeProperties(const std::shared_ptr<arrow::Table>& props) {
-  return UpsertProperties(
-      props, &node_properties_, &part_header_.node_prop_info_list());
+RDGCore::UpsertNodeProperties(
+    const std::shared_ptr<arrow::Table>& props, tsuba::TxnContext* ctx) {
+  KATANA_LOG_DEBUG_ASSERT(ctx != nullptr);
+  auto written_prop_names = KATANA_CHECKED(UpsertProperties(
+      props, &node_properties_, &part_header_.node_prop_info_list()));
+  // store write properties into transaction context
+  ctx->InsertNodePropertyWrite(std::move(written_prop_names));
```

Comment:
I'm not sure everyone agrees, but I feel like the parameter should be `std::vector<std::string>&&` if you expect clients to `std::move` into it.  But Tyler does this std::move as an optimization also.

Reply:
I think they're the same in terms of efficiency. The smart pointer passes the ownership, which is equivalent to passing callee the raw pointer.